modules/core/src/main/java/org/apache/ignite/IgniteSystemProperties.java

```diff
@@ -840,7 +841,14 @@
      * @see CacheConfiguration#readFromBackup
      */
     public static final String IGNITE_READ_LOAD_BALANCING = "IGNITE_READ_LOAD_BALANCING";
-  
+
+    /**
+     * Property allowing to serialize a singleton collections obtained by {@link Collections#singleton(Object)}
+     * and {@link Collections#singletonMap(Object, Object)}.
+     */
+    public static final String IGNITE_SUPPORT_SINGLETON_COLLECTION_SERIALIZATION = "IGNITE_SUPPORT_SINGLETON_COLLECTION_SERIALIZATION";
```

Comment:
Why does it have to be a property and not the default? What would break otherwise?

Reply:
To keep persistance compatibility between 2.4 and 2.5